lib/puppet/application/agent.rb

```diff
@@ -71,6 +71,10 @@ def preinit
     options[:waitforcert] = arg.to_i
   end
 
+  option("--job-id ID") do |arg|
+    options[:job_id] = arg
```

Comment:
Yeah, UTF-8 should be fine. The contract is that it's an opaque string, though in practice it's just an integer [stringified] today and perhaps some day a uuid.